### PR TITLE
Merchant fulfill#71

### LINKED ISSUE #71: User Story 60, Merchant fulfills part of an order
```
As a merchant
When I visit an order show page from my dashboard
For each item of mine in the order
If the user's desired quantity is equal to or less than my current inventory quantity for that item
And I have not already "fulfilled" that item:
- Then I see a button or link to "fulfill" that item
- When I click on that link or button I am returned to the order show page
- I see the item is now fulfilled
- I also see a flash message indicating that I have fulfilled that item
- My inventory quantity is permanently reduced by the user's desired quantity

If I have already fulfilled this item, I see text indicating such.
```

### Checklist:

- [ ] base code tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #73

depends on #72

### PR DESCRIPTION
**Pull request by stoic-plus**
closes #71 and #98 

* Adds `update_status` and `complete?` instance methods to `Order`
* Adds `decrease_inventory` to `Item`
* Adds `fulfill_order_item` to `OrderItemsController` and route for it (POST)

### CODE REVIEW

File: spec/models/order_spec.rb [hunk 1]
```diff
@@ -247,5 +247,51 @@
...
+
+      expect(o.status).to eq("pending")
+
+      oi_2.fulfilled = true
+      oi_2.save
+      oi_2.reload
+      o.reload
+
+      o.update_status
+      expect(o.status).to eq("complete")
```
Comment (tonyr729) [inline spec/models/order_spec.rb]: If oi_1 was not fulfilled the status would still remain pending. We're all lines covered with this test?
